auth_autologin_via_jwt_cookie/models/ir_http.py

```diff
@@ -0,0 +1,137 @@
+import logging
+import time
+
+import requests
+from jose import jwk, jwt
+from jose.utils import base64url_decode
+
+from odoo import models
+from odoo.http import request
+
+_logger = logging.getLogger(__name__)
+
+
+class IrHttp(models.AbstractModel):
+    _inherit = "ir.http"
+
+    @classmethod
+    def _authenticate(cls, endpoint):
+        if getattr(request, "session", None) and request.session.uid:
+            return super()._authenticate(endpoint)
+
+        try:
+            icp = request.env["ir.config_parameter"].sudo()
+            cookie_name = (
+                icp.get_param("auth_autologin_via_jwt_cookie.jwt_cookie_name") or ""
+            ).strip()
+            jwks_url = (
+                icp.get_param("auth_autologin_via_jwt_cookie.jwks_url") or ""
+            ).strip()
+            userinfo_url = (
+                icp.get_param("auth_autologin_via_jwt_cookie.userinfo_url") or ""
+            ).strip()
+
+            if not (cookie_name and jwks_url and userinfo_url):
+                return super()._authenticate(endpoint)
+
+            token = request.httprequest.cookies.get(cookie_name)
+            if not token:
+                return super()._authenticate(endpoint)
+
+            claims = cls._verify_jwt_with_jwks(token, jwks_url)
+            if not claims:
+                return super()._authenticate(endpoint)
+
+            email = cls._get_email_from_userinfo(userinfo_url, token)
+            if not email:
+                return super()._authenticate(endpoint)
+
+            user = (
+                request.env["res.users"]
+                .sudo()
+                .search(
+                    ["|", ("login", "=", email), ("email", "=", email)],
+                    limit=1,
+                )
+            )
+            if not user or not user.active:
+                return super()._authenticate(endpoint)
+
+            cls._force_login(user)
+        except Exception:
+            _logger.exception(
+                "JWT cookie autologin failed; falling back to default auth."
+            )
+            return super()._authenticate(endpoint)
```

Comment:
that's a very big try block with a very generic Exception, can you try smaller try blocks and maybe refactor in a sub private method to avoid some many `return super()._authenticate(endpoint)`